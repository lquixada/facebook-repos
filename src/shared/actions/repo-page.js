export const REPO_PAGE_REQUESTED = 'REPO_PAGE_REQUESTED'
export const REPO_PAGE_SUCCEEDED = 'REPO_PAGE_SUCCEEDED'
export const REPO_PAGE_FAILED = 'REPO_PAGE_FAILED'

export const loadRepoPage = (repoName) => ({
  type: REPO_PAGE_REQUESTED,
  payload: {
    repoName
  }
})
