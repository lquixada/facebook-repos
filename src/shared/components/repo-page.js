import React from 'react';

import Repo from '../containers/repo';
import Menu from '../containers/menu';
import {Footer} from './footer';
import {Title} from './title';
import {Main, Aside, Content, Header, HeaderContainer, Section, SectionContainer} from './grid';

export const RepoPage = ({match}) => (
  <Main>
    <Header>
      <HeaderContainer>
        <Title />
      </HeaderContainer>
    </Header>

    <Section>
      <SectionContainer>
        <Aside>
          <Menu />
          <Footer />
        </Aside>
        <Content>
          <Repo match={match} />
        </Content>
      </SectionContainer>
    </Section>
  </Main>
);
