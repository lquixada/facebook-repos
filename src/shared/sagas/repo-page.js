import {take, fork} from 'redux-saga/effects';

import {REPO_PAGE_REQUESTED} from '../actions';
import {loadContributors} from './contributors';
import {loadRepo} from './repo';
import {loadReposContributorsCount} from './repos-contributors-count';

export default function* watchLoadRepoPage() {
  while (true) {
    const {payload} = yield take(REPO_PAGE_REQUESTED);

    yield fork(loadReposContributorsCount);
    yield fork(loadRepo, payload.repoName);
    yield fork(loadContributors, payload.repoName);
  }
}
