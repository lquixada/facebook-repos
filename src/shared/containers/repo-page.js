import React from 'react'
import {compose} from 'redux'
import {connect} from 'react-redux'
import {RepoPage} from '../components/repo-page'
import {provideHooks} from '../helpers'
import * as actions from '../actions'

const hooks = {
  fetch: ({params, dispatch}) => dispatch(actions.fetchPage({
    name: 'repo',
    owner: params.owner || 'facebook',
    repoName: params.repo
  }))
}

export class RepoPageContainer extends React.Component {
  componentDidUpdate (prevProps) {
    if (this.hasChanged(prevProps)) {
      this.fetch(this.props)
    }
  }

  fetch ({owner, repo}) {
    this.props.fetchPage({
      name: 'repo',
      owner: owner || 'facebook',
      repoName: repo
    })
  }

  hasChanged (prevProps) {
    return (
      prevProps.owner !== this.props.owner ||
      prevProps.repoName !== this.props.repoName
    )
  }

  render () {
    return (
      <RepoPage repoName={this.props.repoName} />
    )
  }
}

const mapStateToProps = (state, {match}) => ({
  owner: match.params.owner,
  repoName: match.params.repo
})
const composed = compose(provideHooks(hooks), connect(mapStateToProps, actions))
export default composed(RepoPageContainer)
