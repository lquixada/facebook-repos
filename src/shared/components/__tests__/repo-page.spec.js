import React from 'react';
import {shallow} from 'enzyme';

import {RepoPage} from '../repo-page';

describe('<RepoPage />', () => {
  it('matches the snapshot', () => {
    const component = shallow(<RepoPage />);
    expect(component).toMatchSnapshot();
  });
});
