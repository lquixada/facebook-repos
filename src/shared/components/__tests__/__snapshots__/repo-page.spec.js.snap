// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoPage /> matches the snapshot 1`] = `
<styled.main>
  <styled.header>
    <styled.div>
      <Title />
    </styled.div>
  </styled.header>
  <styled.section>
    <styled.div>
      <styled.aside>
        <withRouter(Connect(Menu)) />
        <Footer />
      </styled.aside>
      <styled.article>
        <Route
          component={[Function]}
          path="/r/:repo"
        />
      </styled.article>
    </styled.div>
  </styled.section>
</styled.main>
`;
